tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -516,8 +547,19 @@ private void replayLogs(Collection<TransactionLog> logs) {
           editCnt++;
           switch (edit.getState()) {
             case INPROGRESS:
+              long expiration = edit.getExpiration();
+              TransactionType type = edit.getType();
+              // Check if transaction needs to be migrated to have expiration and type. Previous version of 
+              // long running transactions were represented with expiration time as -1.
+              // This can be removed when we stop supporting TransactionEditCodecV2.
+              if (expiration < 0) {
+                expiration = getTxExpiration(defaultLongTimeout);
```

Comment:
Same comment here on the expiration time.  The expiration should be based on the transaction, not when the WAL was replayed.